### ISSUE TITLE
new install of v2.0 crashes on Ubuntu 16.04

### ISSUE THREAD

**Issue by Chrescht** [duplicati/duplicati#1710]
I installed Duplicati 2.0 on Ubuntu 16.04 (64bits) as per [the instructions](http://www.duplicati.com/howtos/how-to-install-and-run-duplicati-2-0-preview) but wit no luck:

`mono /opt/duplicati/Duplicati.GUI.TrayIcon.exe`

```
Failed to load Mono.Data.Sqlite.SqliteConnection, reverting to built-in.
No LSB modules are available.
Crash! 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.TypeInitializationException: The type initializer for 'Duplicati.Server.Serialization.Serializer' threw an exception. ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
File name: 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
  --- End of inner exception stack trace ---
  at Duplicati.GUI.TrayIcon.HttpServerConnection.PerformRequestInternal[T] (System.String method, System.String endpoint, System.Collections.Generic.Dictionary`2 queryparams) <0x420fc430 + 0x00963> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection.PerformRequest[T] (System.String method, System.String urlfragment, System.Collections.Generic.Dictionary`2 queryparams) <0x420fbe40 + 0x000eb> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection.UpdateStatus () <0x420fbcc0 + 0x0003b> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.HttpServerConnection..ctor (System.Uri server, System.String password, Boolean saltedpassword) <0x420f9c50 + 0x0035b> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.RealMain (System.String[] _args) <0x42028000 + 0x0084f> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x42026480 + 0x000b7> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x42026480 + 0x00123> in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) <0x41fb5ef0 + 0x0002a> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) <0x420260d0 + 0x00089> in <filename unknown>:0 

Unhandled Exception:
System.TypeInitializationException: The type initializer for 'Duplicati.Server.Serialization.Serializer' threw an exception.

Server stack trace: 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) <0x420260d0 + 0x00357> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager+<RunFromMostRecent>c__AnonStorey7.<>m__0 () <0x42026090 + 0x0001f> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater (AutoUpdateStrategy defaultstrategy, System.Action wrappedFunction) <0x42025ac0 + 0x00342> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x42011380 + 0x0025b> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x41fc1c00 + 0x00053> in <filename unknown>:0 
  at (wrapper managed-to-native) System.AppDomain:ExecuteAssembly (System.AppDomain,System.Reflection.Assembly,string[])
  at System.AppDomain.ExecuteAssemblyInternal (System.Reflection.Assembly a, System.String[] args) <0x41fc1970 + 0x00054> in <filename unknown>:0 
  at System.AppDomain.ExecuteAssemblyByName (System.String assemblyName, System.String[] args) <0x41fc17d0 + 0x0003c> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper xdomain-dispatch) System.AppDomain:ExecuteAssemblyByName (object,byte[]&,byte[]&,string,string[])

Exception rethrown at [0]: 
 ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
File name: 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
  --- End of inner exception stack trace ---
  at (wrapper xdomain-invoke) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x40770850 + 0x006f9> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x40737d50 + 0x00053> in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.TypeInitializationException: The type initializer for 'Duplicati.Server.Serialization.Serializer' threw an exception.

Server stack trace: 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) <0x420260d0 + 0x00357> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager+<RunFromMostRecent>c__AnonStorey7.<>m__0 () <0x42026090 + 0x0001f> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater (AutoUpdateStrategy defaultstrategy, System.Action wrappedFunction) <0x42025ac0 + 0x00342> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x42011380 + 0x0025b> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x41fc1c00 + 0x00053> in <filename unknown>:0 
  at (wrapper managed-to-native) System.AppDomain:ExecuteAssembly (System.AppDomain,System.Reflection.Assembly,string[])
  at System.AppDomain.ExecuteAssemblyInternal (System.Reflection.Assembly a, System.String[] args) <0x41fc1970 + 0x00054> in <filename unknown>:0 
  at System.AppDomain.ExecuteAssemblyByName (System.String assemblyName, System.String[] args) <0x41fc17d0 + 0x0003c> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper xdomain-dispatch) System.AppDomain:ExecuteAssemblyByName (object,byte[]&,byte[]&,string,string[])

Exception rethrown at [0]: 
 ---> System.IO.FileNotFoundException: Could not load file or assembly 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies.
File name: 'System.Runtime.Serialization, Version=3.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'
  --- End of inner exception stack trace ---
  at (wrapper xdomain-invoke) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x40770850 + 0x006f9> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x40737d50 + 0x00053> in <filename unknown>:0 
```
##

**Comment by Chrescht** [comment]
Linked issue #1211 for {L}Ubuntu 14.04.

I can confirm Duplicati 2.0 runs without errors after installing the meta-package mono-complete.
Yes it's overkill, there are 137 new packages and 135MB of disk space.

**Comment by anarcat** [comment]
i have a similar error running Duplicati in the latest Debian Jessie release, with mono-complete installed:

```
angela:~$ duplicati
No LSB modules are available.
Server has started and is listening on 127.0.0.1, port 8200

** (Duplicati.GUI.TrayIcon:27251): WARNING **: Unknown icon format, bitcount = 21060, compression = 65536
Request error: Thread was being aborted
Crash! 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]
  at System.Drawing.GDIPlus.CheckStatus (Status status) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.CreateFromHandle (IntPtr handle) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.LoadFromStream (System.IO.Stream stream, Boolean keepAlive) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.FromStream (System.IO.Stream stream) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.LoadImage (System.String filename) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.get_StatusMenuIcon () [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper.GetIcon (MenuIcons icon) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper..ctor (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.CreateMenuItem (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.BuildMenu () [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.Init (System.String[] args) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.Init (System.String[] args) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.RealMain (System.String[] _args) [0x00000] in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) [0x00000] in <filename unknown>:0 

Unhandled Exception:
System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]

Server stack trace: 
  at System.Drawing.GDIPlus.CheckStatus (Status status) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.CreateFromHandle (IntPtr handle) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.LoadFromStream (System.IO.Stream stream, Boolean keepAlive) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.FromStream (System.IO.Stream stream) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.LoadImage (System.String filename) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.get_StatusMenuIcon () [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper.GetIcon (MenuIcons icon) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper..ctor (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.CreateMenuItem (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.BuildMenu () [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.Init (System.String[] args) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.Init (System.String[] args) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.RealMain (System.String[] _args) [0x00000] in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 

Exception rethrown at [0]: 

  at (wrapper xdomain-invoke) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]

Server stack trace: 
  at System.Drawing.GDIPlus.CheckStatus (Status status) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.CreateFromHandle (IntPtr handle) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.LoadFromStream (System.IO.Stream stream, Boolean keepAlive) [0x00000] in <filename unknown>:0 
  at System.Drawing.Image.FromStream (System.IO.Stream stream) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.LoadImage (System.String filename) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.get_StatusMenuIcon () [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper.GetIcon (MenuIcons icon) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper..ctor (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.CreateMenuItem (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.BuildMenu () [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.Init (System.String[] args) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.Init (System.String[] args) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.RealMain (System.String[] _args) [0x00000] in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00000] in <filename unknown>:0 

Exception rethrown at [0]: 

  at (wrapper xdomain-invoke) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) [0x00000] in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) [0x00000] in <filename unknown>:0 
```

**Comment by G4Zz0L1** [comment]
I already have mono-complete installet (Mint 18) but there's no difference...

**Comment by kenkendk** [comment]
I can reproduce on Linux Mint, and I cannot fix it with mono-complete either.

I think the problem is that we are using some icon files that GDI does not understand.
I will make some new files in PNG format for use on Linux.

**Comment by G4Zz0L1** [comment]
Nope, I've the same identical error as before

`Server has started and is listening on 127.0.0.1, port 8200`

```
** (Duplicati.GUI.TrayIcon:12116): WARNING **: Unknown icon format, bitcount = 21060, compression = 65536
Request error: 
Crash! 
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]
  at System.Drawing.GDIPlus.CheckStatus (Status status) <0x42170940 + 0x00167> in <filename unknown>:0 
  at System.Drawing.Image.CreateFromHandle (IntPtr handle) <0x421707a0 + 0x00027> in <filename unknown>:0 
  at System.Drawing.Image.LoadFromStream (System.IO.Stream stream, Boolean keepAlive) <0x4214c540 + 0x00033> in <filename unknown>:0 
  at System.Drawing.Image.FromStream (System.IO.Stream stream) <0x4214c520 + 0x0000f> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.LoadImage (System.String filename) <0x4214c290 + 0x000ef> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.ImageLoader.get_StatusMenuIcon () <0x4214c150 + 0x00013> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper.GetIcon (MenuIcons icon) <0x4214bd10 + 0x00163> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner+MenuItemWrapper..ctor (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) <0x421483a0 + 0x0015b> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.CreateMenuItem (System.String text, MenuIcons icon, System.Action callback, IList`1 subitems) <0x42148320 + 0x0004f> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.BuildMenu () <0x42148000 + 0x000af> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.TrayIconBase.Init (System.String[] args) <0x42147e70 + 0x00017> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.GtkRunner.Init (System.String[] args) <0x42145cb0 + 0x00027> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.RealMain (System.String[] _args) <0x41fc47b0 + 0x00bc1> in <filename unknown>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x41fc31c0 + 0x000b7> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at System.Reflection.MonoMethod.Invoke (System.Object obj, BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) <0x41fc31c0 + 0x00123> in <filename unknown>:0 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) <0x41ef8ff0 + 0x0002a> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) <0x41fc2e10 + 0x00089> in <filename unknown>:0 

Unhandled Exception:
System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]

Server stack trace: 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) <0x41fc2e10 + 0x00357> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager+<RunFromMostRecent>c__AnonStorey7.<>m__0 () <0x41fc2dd0 + 0x0001f> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater (AutoUpdateStrategy defaultstrategy, System.Action wrappedFunction) <0x41fc2800 + 0x00342> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x41fae000 + 0x00273> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x41f06df0 + 0x00053> in <filename unknown>:0 
  at (wrapper managed-to-native) System.AppDomain:ExecuteAssembly (System.AppDomain,System.Reflection.Assembly,string[])
  at System.AppDomain.ExecuteAssemblyInternal (System.Reflection.Assembly a, System.String[] args) <0x41f06b60 + 0x00054> in <filename unknown>:0 
  at System.AppDomain.ExecuteAssemblyByName (System.String assemblyName, System.String[] args) <0x41f069c0 + 0x0003c> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper xdomain-dispatch) System.AppDomain:ExecuteAssemblyByName (object,byte[]&,byte[]&,string,string[])

Exception rethrown at [0]: 

  at (wrapper xdomain-invoke) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x40a633f0 + 0x0072c> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x409fcd50 + 0x00053> in <filename unknown>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.ArgumentException: A null reference or invalid value was found [GDI+ status: InvalidParameter]

Server stack trace: 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunMethod (System.Reflection.MethodInfo method, System.String[] args) <0x41fc2e10 + 0x00357> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager+<RunFromMostRecent>c__AnonStorey7.<>m__0 () <0x41fc2dd0 + 0x0001f> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.WrapWithUpdater (AutoUpdateStrategy defaultstrategy, System.Action wrappedFunction) <0x41fc2800 + 0x00342> in <filename unknown>:0 
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x41fae000 + 0x00273> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x41f06df0 + 0x00053> in <filename unknown>:0 
  at (wrapper managed-to-native) System.AppDomain:ExecuteAssembly (System.AppDomain,System.Reflection.Assembly,string[])
  at System.AppDomain.ExecuteAssemblyInternal (System.Reflection.Assembly a, System.String[] args) <0x41f06b60 + 0x00054> in <filename unknown>:0 
  at System.AppDomain.ExecuteAssemblyByName (System.String assemblyName, System.String[] args) <0x41f069c0 + 0x0003c> in <filename unknown>:0 
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper xdomain-dispatch) System.AppDomain:ExecuteAssemblyByName (object,byte[]&,byte[]&,string,string[])

Exception rethrown at [0]: 

  at (wrapper xdomain-invoke) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at (wrapper remoting-invoke-with-check) System.AppDomain:ExecuteAssemblyByName (string,string[])
  at Duplicati.Library.AutoUpdater.UpdaterManager.RunFromMostRecent (System.Reflection.MethodInfo method, System.String[] cmdargs, AutoUpdateStrategy defaultstrategy) <0x40a633f0 + 0x0072c> in <filename unknown>:0 
  at Duplicati.GUI.TrayIcon.Program.Main (System.String[] args) <0x409fcd50 + 0x00053> in <filename unknown>:0
```

**Comment by kenkendk** [comment]
@G4Zz0L1 what version is reported if you run `duplicati-cli`
Is this a re-install (i.e. from a downloaded .deb/.zip) or an updated install?

**Comment by oceanos59** [comment]
Hi, my issue has solved, i don t have tray icon issue anymore and duplicati works very well with version 2.0.1.29 . Thank you very much ! You are awesome !

**Comment by G4Zz0L1** [comment]
http://www.duplicati.com/              Version:  - 2.0.1.27_canary_2016-10-14

It's strange, I've purged it and reinstalled it from the deb on the release. Why it's still the .27?

I'll try to repurge and check for residual files, maybe it's only that.

**Comment by G4Zz0L1** [comment]
Nope, same thing. I've purged the package, and I've checked that nothing's left into /usr/lib, but when I reinstall the package it yelds that version again. It's normal?

**Comment by kakoskin1** [comment]
@G4Zz0L1: If you have used autoupdater, you may have to remove version autoupdater installed. On Fedora 24 I had to remove directory ~/.local/share/Duplicati/ after installing 2.0.1.29 rpm. 

See: https://github.com/duplicati/duplicati/issues/2061

**Comment by G4Zz0L1** [comment]
Hmm, right. I'll try the next week, the problem it's only on my workstation, here at home everything's fine. I'll let you know if it works, meanwhile thanks for the help!

**Comment by kenkendk** [comment]
There is an "updates" folder in the `~/.local/share/Duplicati/` that you can remove.
This is very likely the #2061 problem.

**Comment by G4Zz0L1** [comment]
Yep, that was the problem, but the folder to remove was on ~/.config/Duplicati. Now it works fine.

Thanks for the help!

**Comment by kenkendk** [comment]
Great!

I have fixed issue #2061 too, it will be in the next canary.
